homeassistant/components/rfxtrx.py

```diff
@@ -77,9 +77,13 @@ def _valid_device(value, device_type):
         if not len(key) % 2 == 0:
             key = '0' + key
 
-        if get_rfx_object(key) is None:
-            raise vol.Invalid('Rfxtrx device {} is invalid: '
-                              'Invalid device id for {}'.format(key, value))
+        try:
+            if get_rfx_object(key) is None:
+                raise vol.Invalid('Rfxtrx device {} is invalid: Invalid'
+                                  ' device id for {}'.format(key, value))
+        except ImportError:
+            _LOGGER.warning('Restart Home Assistant to'
```

Comment:
Shouldn't `vol.Invalid` be raised? Otherwise a possible invalid configuration could be passed as valid. What consequences would that have for downstream code?

Reply:
If we raise an error here, I think the rfxtrx lib never will be installed?
If the key is invalid the first time HA starts, the rfxtrx component should still load and install the rfxtrx lib, but the switch/sensor with an error will not be loaded.